FFXIVClientStructs/FFXIV/Client/UI/Agent/AgentItemDetail.cs

```diff
@@ -8,5 +10,31 @@ namespace FFXIVClientStructs.FFXIV.Client.UI.Agent;
 [Inherits<AgentInterface>]
 [StructLayout(LayoutKind.Explicit, Size = 0x1B8)]
 public unsafe partial struct AgentItemDetail {
+    public enum ItemDetailKind : byte {
+        ChatItem = 1,      // all items linked in chat, except event items
+        InventoryItem = 2, // all(?) items outside of chat, including event items
+        ChatEventItem = 8  // event items linked in chat
+    }
+
+    [FieldOffset(0x118)] public ItemDetailKind ItemKind;
+    // Set to the item ID when hovering an item in the chat, otherwise it seems
+    // to be different for each inventory. Doesn't appear to have any relation
+    // to InventoryType.
+    // -  4: EquippedItems
+    // -  7: KeyItems
+    // - 48: Inventory1
+    // - 49: Inventory2
+    // - 50: Inventory3
+    // - 51: Inventory4
+    // - 69: SaddleBag1
+    // - 70: SaddleBag2
+    // - 71: PremiumSaddleBag1
+    // - 72: PremiumSaddleBag2
+    [FieldOffset(0x11C)] public uint TypeOrId;
+    [FieldOffset(0x120)] public uint Index; // The index of the item in the inventory. 0 for chat, ??? for KeyItems.
+    [FieldOffset(0x128)] public byte Flag128; // Related to checking the inventory
```

Comment:
Preferably don't name flags with their offset as the name and instead do `Flag1`, `Flag2` and so on